docs/source/ddp_comm_hooks.rst

```diff
@@ -0,0 +1,64 @@
+DDP Communication Hooks
+=======================
+
+DDP communication hook is a generic interface to control how to communicate
```

Comment:
Might be good to explicitly note that these hooks are used in conjunction with DistributedDataParallel in the first paragraph and note that it will override the allreduce.

Reply:
Updated the first sentence as:

DDP communication hook is a generic interface to control how to communicate gradients across workers and override the vanilla allreduce in `DistributedDataParallel`.